primo-explore/custom/01MIT_INST-TACOS/js/custom.js

```diff
@@ -14,8 +42,56 @@
       '<div id="title-mit"><a href="https://mit.primo.exlibrisgroup.com/discovery/search?vid=01MIT_INST:MIT&lang=en">Search Our Collections</a></div>',
   });
   app.component("prmSearchBarAfter", {
-    template: alertBannerTemplate,
+    template: `<tacos></tacos>${alertBannerTemplate}`,
+  });
+  app.component("tacos", {
+    controller: function tacosController($scope, tacosService, $stateParams) {
+      // $stateParams seems to behave like an angularJS service, but I'm not sure
+      // where in the Primo VE source code it is defined...
+      // We are able to inject it in the controller and then get access to the .query property
+      // which is the raw search string(s) we want to send to TACOS
+
+      var vm = this;
+
+      function parseSearchQuery(searchQuery) {
+        // Returns searchQuery if it is a string.
+        // If searchQuery is an array of strings, joins the strings and returns the result
+
+        if (typeof searchQuery === "string") { // handle a string (simple search)
+          return searchQuery;
+        }
+        return searchQuery.join(" "); // handle an array of strings (advanced search)

```

Comment:
I worry a little about using a space as a separator in this way, as the search query itself will likely also be a phrase involving spaces. I don't mind doing it this way for now, while we're exploring our options - but ultimately I think it could introduce uncertainty within TACOS to not have a more explicit structure. My instinct is to encode the searchQery construct as JSON.

Reply:
yeah, space was just a place holder, it could be anything. I like the idea of encoding this as JSON. The implementation would depend on what TACOS expects to recieve.